### PR TITLE
Update invite users to community modal

### LINKED ISSUE #6382: Update invite users to community modal
figma: https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?node-id=2927%3A343592

Screen Shot 2022-07-07 at 3 22 29 PM

Screen Shot 2022-07-07 at 3 23 27 PM

### PR DESCRIPTION
**Pull request by MishkaRogachev**
Close #6382
Needs https://github.com/status-im/status-go/pull/2776
Needs https://github.com/status-im/StatusQ/pull/811

### What does the PR do

New invite to community popup with two pages

### Affected areas

Communities

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

https://user-images.githubusercontent.com/2522130/181515217-db7e01e9-d205-4a40-b894-4839bdccfc2d.mov

### CODE REVIEW

File: ui/app/AppLayouts/Chat/controls/Contact.qml [hunk 1]
```diff
@@ -33,8 +33,6 @@ Rectangle {
 
     visible: isVisible && (isContact || isUser)
     height: visible ? 64 : 0
-    anchors.right: parent.right
-    anchors.left: parent.left
```
Comment (jrainville) [inline ui/app/AppLayouts/Chat/controls/Contact.qml:37]: Just to be sure, you checked all places where `Contact` was used and added the width or anchors there instead?

File: ui/app/AppLayouts/Chat/panels/communities/CommunityProfilePopupInviteMessagePanel.qml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    StatusBaseText {
+        text: qsTr("Invitation Message")
+        font.pixelSize: 13
+    }
+
+    StatusInput {
+        id: messageInput
+        charLimit: d.maxMsgLength
+        text: qsTr("Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque interdum rutrum sodales. Nullam mattis fermentum libero, non volutpat.")
```
Comment (jrainville) [inline ui/app/AppLayouts/Chat/panels/communities/CommunityProfilePopupInviteMessagePanel.qml:41]: Remove the lorem ipsum.
Do we need a placeholder instead?

File: ui/app/AppLayouts/Chat/popups/community/CommunityProfilePopup.qml [hunk 3]
```diff
@@ -123,7 +123,7 @@ StatusModal {
             height: !visible ? 0 : implicitHeight
             enabled: root.contentItem.currentItem.contactListSearch !== undefined && root.contentItem.currentItem.contactListSearch.pubKeys.length > 0
             onClicked: {
-                root.contentItem.currentItem.sendInvites(root.contentItem.currentItem.contactListSearch.pubKeys)
+                root.contentItem.currentItem.sendInvites(root.contentItem.currentItem.contactListSearch.pubKeys, "") // NOTE: empty message
```
Comment (jrainville) [inline ui/app/AppLayouts/Chat/popups/community/CommunityProfilePopup.qml:126]: Shouldn't this send the message?

File: ui/app/AppLayouts/Chat/popups/community/InviteFriendsToCommunityPopup.qml [hunk 4]
```diff
@@ -1,80 +1,90 @@
...
     }
 
-    margins: 32
-    height: 550
+    stackTitle: qsTr("Invite Contacts to %1").arg(community.name)
+    implicitWidth: 640
+    implicitHeight: 700
 
-    header.title: qsTr("Invite friends")
+    // header.title: qsTr("Invite friends")
```
Comment (jrainville) [inline ui/app/AppLayouts/Chat/popups/community/InviteFriendsToCommunityPopup.qml:49]: leftover

**Review by jrainville — CHANGES_REQUESTED** [review]
Good work!

Requesting changes because some small things need to be addressed before merging.

File: ui/app/AppLayouts/Chat/controls/Contact.qml (hunk 1 above)
Comment (MishkaRogachev) [reply ui/app/AppLayouts/Chat/controls/Contact.qml:37]: Yes, i've checked other places with Contacts

File: ui/app/AppLayouts/Chat/panels/communities/CommunityProfilePopupInviteMessagePanel.qml (hunk 2 above)
Comment (MishkaRogachev) [reply ui/app/AppLayouts/Chat/panels/communities/CommunityProfilePopupInviteMessagePanel.qml:41]: There is [comment](feat/issue-6382-update-invite-users-to-community-modal) in figma about pre-filled message. for now i'll add placeholder "The message a contact will get with community invitation"

File: ui/app/AppLayouts/Chat/popups/community/CommunityProfilePopup.qml (hunk 3 above)
Comment (MishkaRogachev) [reply ui/app/AppLayouts/Chat/popups/community/CommunityProfilePopup.qml:126]: IDK, @jrainville i think this part of code does not being used any more

**Comment by MishkaRogachev** [comment]
@glitchminer when you start testing this feature can you please message me: status-go should be updated on the both sender and receiver clients

**Review by saledjenic — APPROVED** [review]

**Review by jrainville — APPROVED** [review]

**Comment by glitchminer** [comment]
## Issues Found

1. OPEN. Figma shows only a list of mutual contacts and a search for those contacts. Current input allows for non-contacts to be added.
---

2. OPEN. Default text in search box does not show search icon and the words "Search contacts"

   Expected:
    image

   Actual:
    image
---

3. OPEN. Search button is hidden due to width of dialog
   image
---

4. OPEN. Activity status is missing
   > Understood to be dependant on a separate PR.
---

5. OPEN. Usernames in list do not show chat key.
   > Understood to be dependant on a separate PR (Permutations of usernames).
---

6. OPEN. Checkboxes do not align in modal as per Figma

   Expected
    image
 
   Actual
   image
---

7. OPEN. Share button does not align in modal as per Figma

   Expected
   image

   Actual
   image
---

8. OPEN. No validation on "Next" button. It is possible to proceed without selecting any contacts.
---

9. OPEN. The invite menu is missing

   Expected
   image

   > Understood to be dependant on a separate PR.

---

10. OPEN. Heading and character count are not all on one line
   Expected
   image
   
    Actual
   image
---

11. OPEN. Button reads "Invite" instead of "Send Invites"
    
   Expected
   
image

---

12. OPEN. Default invitation message is shortened
   
    Actual
    image
---

13. OPEN. Contacts highlight on hover even though there is no functionality
   
    Actual
   image
---

14. OPEN. Contact names are shortened more than in Figma. 13 chars instead of 18.
   
   Expected
   image

   Actual
    image
---

15. OPEN. Double line on bottom of modal
   
    Expected
   image

    Actual
   image
---

16. OPEN. Heading does not support maximum length of community name (30 chars.)
   Actual
   Title cuts short at 25 chars.
   image
---

END.

**Comment by MishkaRogachev** [comment]
@glitchminer, there is few fixes:
1 - fixed
2 - fixed
3 - fixed
4 - separate issue
5 - separate issue
6 - fixed
7 - it’s widely used component, separated issue
8 - fixed
9 - separated issue
10 - fixed
11 - fixed
12 - fixed
13 - fixed 
14 - it depends on component’s width, increased 
15 - fixed 
16 - fixed with increased  width 
17 - fixed

**Comment by glitchminer** [comment]
@MishkaRogachev retested. Item 17 failed again, please can you take a look? https://github.com/status-im/status-desktop/pull/6655#issuecomment-1205469876

**Comment by MishkaRogachev** [comment]
> @MishkaRogachev retested and item 17 failed again, please can you take a look? [#6655 (comment)](https://github.com/status-im/status-desktop/pull/6655#issuecomment-1205469876). Other issues passed and were updated. Thanks!

Thanks, i'll check it tomorrow

**Review by glitchminer — APPROVED** [review]
Tested and approved.

**Pull request merged**